packages/block-editor/src/components/inserter/hooks/use-insertion-point.js

```diff
@@ -145,18 +167,21 @@ function useInsertionPoint( {
 			getSelectedBlock,
 			replaceBlocks,
 			insertBlocks,
-			destinationRootClientId,
-			destinationIndex,
+			blockInsertionRootClientId,
+			blockInsertionIndex,
 			onSelect,
 			shouldFocusBlock,
 			selectBlockOnInsert,
+			setLastFocus,
 		]
 	);
 
 	const onToggleInsertionPoint = useCallback(
 		( show ) => {
 			if ( show ) {
-				showInsertionPoint( destinationRootClientId, destinationIndex );
+				showInsertionPoint( destinationRootClientId, destinationIndex, {
+					block: blockToInsert,
```

Comment:
I assume we need to pass the block so that we can change the target if necessary. Can't be change `destinationRootClientId` and `destinationIndex` so we avoid modifying `showInsertionPoint`?